jQuery/ajax.js

```diff
@@ -21,6 +33,50 @@ function getUniqueValuesForField(field, citizens) {
 
 function populateSelectWithOptions(select, options) {
   options.forEach(function(option) {
-    $(select).append($("<option>", { value: option, html: option }));
+    $(select).append($('<option>', { value: option, html: option }));
+  });
+}
+
+function gatherParams() {
```

Comment:
This function is stateful. Which is fine, if you're okay with side effects sometimes. 

Here's how I'd write this.:

```js
function state(form) {

  // This function is now stateless.
  var character = form.find("#character-select"),
      hometown  = form.find("#hometown-select),
      size      = form.find("#size-select);

  return [
    { field: "character", value: character.val() },
    { field: "hometown", value: hometown.val() },
    { field: "size", value: size.val() }
  ]
}
```

The function is now "stateless" because it takes a reference to a DOM node via jQuery and uses that node to begin finding the elements of the form it needs. 

So in essence you would pass in the jquery selected menu to get all the fields like so:

```js

state($(".menu"))
//or
state($(selectors.menu));

}